sherpa/ui/utils.py

```diff
@@ -110,6 +108,142 @@ def get_plot_prefs(plotobj):
             raise AttributeError("plot object has no preferences") from None
 
 
+def _get_filter(data):
+    """Report the filter for report_filter_change.
+
+    This takes care of calling get_filter with the correct
+    arguments (needed in case anyone wants to use the string
+    in a call to notice/ignore), and it handles the case of
+    the case of all-data-being-filtered leading to an error
+    for certain data types.
+
+    Parameters
+    ----------
+    data : sherpa.data.Data1D instance
+
+    Returns
+    -------
+    msg : str or None
+        The filter expression, with "all-data-filtered" indicated by
+        the empty string, or None when there was an error (which
+        shouldn't happen, but we don't want to fail a notice or ignore
+        call when reporting the change).
+
+    """
+
+    # See #1430 for a discussion on why we have different behavior
+    # when all the data is excluded: it should be unified.
+    #
+    # We follow the approach used for the DataIMG case be relying on
+    # the mask attribute to determine what the output should
+    # be. Unfortunately this is only helpful for the "all data
+    # excluded" case, as we do not have the equivalent of "Field()" to
+    # indicate "use all the data", and we can not remove the DataErr
+    # exception check from get_filter, just in case.
+    #
+    if data.mask is False:
+        return ""
+
+    try:
+        return data.get_filter(delim=':', format='%g')
+    except DataErr as exc:
+        # It may be possible that this happens - e.g. that data.mask
+        # is a ndarray of all False vaues - but it is not 100%
+        # obvious.  This is left in just in case.
+        #
+        if str(exc) == "mask excludes all data":
+            return ""
+
+        # It's not clear if this can happen, but assume it can.
+        #
+        return None
+
+    except TypeError:
+        # This can happen when the dataset is Data2D, as get_filter
+        # does not accept kwargs, so we skip over it (as the handling
+        # of filters for Data2D is not clear).
+        #
+        return None
+
+    except IndexError:
+        # This is a known failure case with ignore_bad handling of
+        # DataPHA.
+        #
+        return None
+
+
+def report_filter_change(idstr, ofilter, nfilter, xlabel=None):
```

Comment:
```suggestion
def report_filter_change(idstr, ofilter, nfilter, xlabel=""):
```
Why make a special case with `None` here? Would it not be easier to simply default to an empty string, then there is no need ot document the None case, and no need to deel with in in line 216-220. (Simply use `"ostr += f"no data -> {nfilter} {label}"` later and if there is no label there is an extra " " at the end. Would that be a problem?

Either way, I'm writing here longer than it takes to do either of those, so feel free to ignore.

Reply:
So, "foo {xlabel} bar" would lead to two spaces if `xlabel` were the empty string. Technically we can handle this with checking for the empty string, but I like the Optional[string] behavior here, as it is not overloading the meaning of what an empty string is. It does rely on the implicit contract that the `Data` `xlabel` value is always set (which it is with our existing data hierarchy).

Reply:
I think this is a stylistic choice so I've not changed it. We can re-visit if we decide to extend/enhance/change the code in the future.